test/capi/capi_utils.hpp

```diff
@@ -152,6 +157,54 @@ static tc_sframe* make_sframe_double(const std::vector<std::pair<std::string, st
     return sf;
   }
 
-
+static bool check_equality_gl_sframe(
+  const turi::gl_sframe sf_gl, const turi::gl_sframe ref_gl) {
+
+  size_t num_columns_sf = sf_gl.num_columns();
+  size_t num_columns_ref = ref_gl.num_columns();
+
+  if (num_columns_sf == num_columns_ref) {
+    std::vector<std::string> column_names_sf = sf_gl.column_names();
+    std::vector<std::string> column_names_ref = ref_gl.column_names();
+    for (size_t column_index=0; column_index < num_columns_sf; column_index++) {
+      // go through all columns and check for sarray equality one by one
+      
+      turi::gl_sarray column_sf = sf_gl.select_column(
+        column_names_sf[column_index]);
+      turi::gl_sarray column_ref = ref_gl.select_column(
+        column_names_ref[column_index]);
+      bool is_float_flag;
+
+      if (column_sf.dtype() != column_ref.dtype()) return false;
+      if (column_sf.dtype() == turi::flex_type_enum::FLOAT) is_float_flag = true;
+
+      for (size_t i = 0; i < column_sf.size(); i++) {
+        if (column_sf[i].get_type() == turi::flex_type_enum::UNDEFINED 
+          || column_ref[i].get_type() == turi::flex_type_enum::UNDEFINED) {
+          if (column_sf[i] != column_ref[i]) return false;
+          else continue;
+        }
+        if (is_float_flag) {
+          if ((std::isnan(column_sf[i].get<turi::flex_float>())) 
+            && (std::isnan(column_ref[i].get<turi::flex_float>()))) {
+            continue;
+          } else {
```

Comment:
Instead of an else block, you could simply put the code below the if block, since the if block will never allow control flow to proceed there. This would avoid a level of indentation.

Reply:
Yup, I'll do that